source/_docs/z-wave/adding.markdown

```diff
@@ -3,81 +3,58 @@ title: "Z-Wave Devices - Adding and Removing"
+2. If you can't remove it from the old controller, **try removing the device using Home Assistant**. Follow the instructions in [Removing Devices](#removing-devices); even though it hasn't been added to Home Assistant yet, most Z-Wave devices will respond to exclusion mode from *any* controller. Then try adding the device to Home Assistant again. (Secure devices with a theft protection feature may not allow this.)
+3. If removing the device doesn't help, **try a factory reset** as a last resort. Check your device's manual for instructions.
 
-<div class='note warning'>
-Secure devices require additional bandwidth, and too many secure devices can slow down your Z-Wave network. We recommend only using secure inclusion for devices that require it, such as locks.
-</div>
+### Range Issues
 
-## Removing Devices
+Normally you can add and remove devices as long as they are within range of any Z-Wave Plus device in your network, using a feature called [network-wide inclusion](https://z-wavealliance.org/z-wave_plus_certification/).
 
-To remove (exclude) a Z-Wave device from your system:
+If you are using older non-Z-Wave Plus devices, your device may need to be within range of your [controller](/docs/z-wave/controllers/) to be added or removed. If you cannot move the device (e.g. wall switches), you can try temporarily relocating your controller to be near your device. See the [Z-Wave Alliance documentation on Z-Wave Plus](https://z-wavealliance.org/z-wave_plus_certification/) for more information on how to determine if your devices support Z-Wave Plus.
 
-1. Go to the Z-Wave control panel in the Home Assistant frontend. (Home Assistant > Configuration > Integrations > Z-Wave > Configure)
-2. Click the **Remove Node** button in the *Z-Wave Network Management* card - this will place the controller in exclusion mode
-3. Activate your device to be excluded by following the instructions provided with the device
-4. The device should now be removed, but that won't show until you restart Home Assistant. Look for a confirmation signal on the device if available, or confirm on the Home Assistant logs.
-5. Run a *Heal Network* so all the other nodes learn about its removal
+### Forcibly Removing Devices
 
-If your device isn't responding to this process, possibly because you've factory reset it or it has failed, you can remove it using **Remove Failed Node**. This only works for devices marked as `"is_failed": true`, but you can trick the system into thinking that this the case:
+You can use the **Remove Failed Node** button to remove a device that has failed (e.g. because it is broken). In rare cases, you may want to remove a device that has *not* failed. This is not recommended, but can help in cases where a device has not been recognized as failed.
```

Comment:
```suggestion
You can use the **Remove Failed Node** button to remove a failed device (e.g., because it is broken). In rare cases, you may want to remove a device that has *not* failed. This is not recommended but can help when a device has not been recognized as failed.
```

Reply:
nice edits, done 👍